### PR TITLE
Add option to treat nulls like nans when doing comparisons

### PR DESCRIPTION
**Pull request by shwina**
https://github.com/rapidsai/cudf/issues/7066 introduced a breaking change wherein comparison operations involving `N/A` return `N/A`, rather than True/False, as was being done previously.

While this is the appropriate behaviour and compatible with Pandas nullable types (which are really the equivalent of what cuDF supports), it breaks compatibility with Pandas _non-nullable_ types.

This PR introduces a runtime configuration option `"nulls_compare_like_nans"`. When set to `True`, cuDF treats `N/A` like `nan` when doing comparisons, thus returning True/False rather than `N/A`.

```python
In [1]: import cudf

In [2]: s = cudf.Series([1, 2, None])

In [3]: s > 1
Out[3]:
0    False
1     True
2     <NA>
dtype: bool

In [4]: with cudf._config.option_context("nulls_compare_like_nans", True):
   ...:     print(s > 1)
   ...:
   ...:
0    False
1     True
2    False
dtype: bool
```

### CODE REVIEW

File: python/cudf/cudf/_config/config.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+
+@contextmanager
+def option_context(*args):
+    global _OPTIONS
+    original_options = _OPTIONS.copy()
+    try:
+        for opt, val in zip(args[::2], args[1::2]):
+            if opt not in _OPTIONS:
+                raise ValueError(f"Invalid option {opt}")
```
Comment (shwina) [inline python/cudf/cudf/_config/config.py:16]: Use a. KeyError instead?

File: python/cudf/cudf/_config/config.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            _OPTIONS[opt] = val
```
Comment (shwina) [inline python/cudf/cudf/_config/config.py:17]: Call `set_option` here.

**Comment by shwina** [comment]
rerun tests

**Review by brandon-b-miller — APPROVED** [review]
LGTM.

File: python/cudf/cudf/_config/config.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
+from contextlib import contextmanager
+
+_OPTIONS = {
+    # nulls compare like NaN/NaT in numeric/datetime comparisons
+    "nulls_compare_like_nans": False
+}
+
+
+@contextmanager
+def option_context(*args):
```
Comment (isVoid) [inline python/cudf/cudf/_config/config.py:10]: Is this implementation thread-safe?

**Review by isVoid — COMMENTED** [review]
A small question on code, and: do we need to add docs for:
- How to use from a user's perspective?
- How should internal code path handle different contexts from a developer's perspective?

**Comment by brandon-b-miller** [comment]
> A small question on code, and do we need to add docs for:
> 
> * How to use from a user's perspective?
> * How should internal code path handle different contexts from a developer's perspective?

I think that support for this is still considered experimental. I'd hold off on developing a lot of final docs for now.

File: python/cudf/cudf/_config/config.py (hunk 3 above)
Comment (shwina) [reply python/cudf/cudf/_config/config.py:10]: Yes, since no two threads can be modifying Python objects at the same time, by virtue of the GIL

Comment (shwina) [reply python/cudf/cudf/_config/config.py:10]: You're right -- this is not thread safe. In an offline conversation, you made the very good point that, within this context, one can release the GIL, and another thread can then pick up the wrong option.

**Comment by shwina** [comment]
Closing this PR as using `fillna()` was found to be a reasonable workaround:


```
In [1]: import cudf

In [2]: s = cudf.Series([1, 2, None])

In [3]: (s > 1).fillna(False)
Out[3]:
0    False
1     True
2    False
dtype: bool
``

**Pull request closed without merging**